personal_finances/user/user_auth.py

```diff
@@ -0,0 +1,124 @@
+import base64
+import json
+import logging
+import boto3
+import bcrypt
+import jwt
+import os
+from datetime import datetime, timedelta
+from typing import Any, Tuple
+
+LOGGER = logging.getLogger(__name__)
+
+
+class AuthorizationNotFound(Exception):
+    def __str__(self) -> str:
+        return "AuthorizationNotFoundException"
+
+
+class UserNotFound(Exception):
+    def __str__(self) -> str:
+        return "UserNotFoundException"
+
+
+class UserInvalidPassword(Exception):
+    def __str__(self) -> str:
+        return "UserInvalidPasswordException"
+
+
+def _decode_basic_auth(auth_header: str) -> Tuple[str, str]:
+    try:
+        auth_value = auth_header.replace("Basic ", "")
+        decoded_bytes = base64.b64decode(auth_value)
+        decoded_str = decoded_bytes.decode("utf-8")
+        userId, password = decoded_str.split(":", 1)
+        return userId, password
+    except Exception:
+        raise AuthorizationNotFound()
+
+
+def _get_auth_header(event: dict) -> str:
+    auth_header = str(event.get("headers", {}).get("Authorization", ""))
+
+    if not auth_header:
+        raise AuthorizationNotFound()
```

Comment:
If I read it correctly, the header itself wasn't present here either because:
1. The default Lambda `event: dict` is wrongly formatted
2. Authorization header is not present
3. Authorization header is present with empty contents

It's crucial to differentiate here because the first one is a Server Error (5xx), and the other are Client Errors (4xx). I'd use different exceptions here, suggestion:
1. `InvalidLambdaEventInput`
2. `AuthorizationHeaderNotPresent`
3. `AuthorizationHeaderEmptyContent`